lib/diagnostic.js

```diff
@@ -6,7 +6,18 @@
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
 const min = function (previous, current) {
-
+  // const workingArray = previous
+  // const result = workingArray.reduce(function (previous, current) {
+  //   return callback(previous, current)
+  // })
+  const workingArray = previous.slice()
+  let result = previous
+  for (let i = 0; i < workingArray.length; i++) {
```

Comment:
This would definitely work, however we're simply looking for a function that will be passed in as a callback to a `reduce` function.